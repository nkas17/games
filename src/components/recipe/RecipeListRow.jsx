import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

/**
 * A Row of a recipe
 */
const RecipeListRow = ({ recipe, deleteRecipe }) => (
	<tr>
		<td>&nbsp;</td>
		<td><Link to={`recipe/${recipe.id}`} >{recipe.title}</Link></td>
		<td>{recipe.description}</td>
		<td>
			<input
				type="button"
				value="delete"
				className="btn btn-link"
				onClick={() => deleteRecipe(recipe._id.$oid)}
			/>
		</td>
	</tr>
	);

RecipeListRow.propTypes = {
	/**
	 * the recipe to display
	 */
	recipe: PropTypes.objectOf(PropTypes.any).isRequired,

	/**
	 * function to delete a recipe
	 */
	deleteRecipe: PropTypes.func.isRequired,
};

export default RecipeListRow;
