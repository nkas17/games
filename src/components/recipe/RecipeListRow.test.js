import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import RecipeListRow from './RecipeListRow';

/* eslint-disable no-undef */

const setup = () => (
	{
		recipe: {
			id: 'tacos',
			title: 'tacos',
			category: 'other',
			description: 'Tasty homemade tacos - hard or soft',
			ingredients: [
				{
					id: 'cheese',
					quantity: '8',
					unit: 'oz',
				},
				{
					id: 'ground beef',
					quantity: '1.5',
					unit: 'lbs',
				},
				{
					id: 'tortilla',
					quantity: '1',
					unit: '',
				},
			],
		},
	}
);

const renderWithRouter = node => renderer.create(<Router>{node}</Router>);

describe('RecipeListRow', () => {
	it('renders correctly', () => {
		const tree = renderWithRouter(
			<RecipeListRow
				{...setup()}
			/>).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
