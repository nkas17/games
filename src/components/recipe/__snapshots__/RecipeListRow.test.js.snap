// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeListRow renders correctly 1`] = `
<tr>
  <td>
    Â 
  </td>
  <td>
    <a
      href="recipe/tacos"
      onClick={[Function]}
    >
      tacos
    </a>
  </td>
  <td>
    Tasty homemade tacos - hard or soft
  </td>
</tr>
`;
